controllers/windowsmachine_controller.go

```diff
@@ -337,6 +332,24 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 	return ctrl.Result{}, nil
 }
 
+// deleteMachineIfAllowed deletes the specified Machine if deletion is allowed.
+// Returns a boolean if the deletion attempt should result in a requeuing.
+func (r *WindowsMachineReconciler) deleteMachineIfAllowed(machine *mapi.Machine) (bool, error) {
```

Comment:
s/deleteMachineIfAllowed/deleteMachine

The `r.isAllowedDeletion(machine)` check is intrinsic of the machine deletion flow, no need to expose it in the function name.

Reply:
There is already another function called `deleteMachine` which is called from `deleteMachineIfAllowed` and [other places](https://github.com/openshift/windows-machine-config-operator/blob/7cc2d2972253c64b5869803caa2d1454900ef05d/controllers/windowsmachine_controller.go#L325) where a machine must be deleted.